### PR TITLE
Graduated renderer fixes

### PR DESCRIPTION
**Pull request by nyalldawson**
Supersedes #31681

### CODE REVIEW

File: src/core/qgsunittypes.cpp [hunk 1]
```diff
@@ -1624,6 +1624,7 @@ double QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::VolumeUnit fromUnit, Qg
       return 1.0;
     }
   }
+  return 1.0;
```
Comment (nirvn) [inline src/core/qgsunittypes.cpp]: Is that change intentional?

**Review by nirvn — APPROVED** [review]
Nice cleanup.

File: src/core/qgsunittypes.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/qgsunittypes.cpp]: yeah, it was just throwing up a warning during compiling
